src/features/user/components/settings.tsx

```diff
@@ -0,0 +1,152 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+
+import { useAuth } from '../../../shared/hooks/useAuth';
+
+import { userSchema } from '../utils/validate';
+import { fromZodError } from 'zod-validation-error';
+
+import { DEFAULT_DATA } from '../constants/data';
+import { ResultData } from '../types/user-data';
+
+import { useMutation } from '@tanstack/react-query';
+import { updateUser, queryClient } from '../utils/fetch-data';
+
+import Swal from 'sweetalert2';
+
+export const ProfileSettings = () => {
+    const [formErrors, setFormErrors] = useState( DEFAULT_DATA );
+    const { userData, update } = useAuth()
+    const navigate = useNavigate()
+    const { mutate } = useMutation({
+        mutationFn: updateUser,
+        onSuccess: (data) => {
+            queryClient.invalidateQueries({queryKey: ['users']})
+            update(data)
+            Swal.fire({
+                title: 'Success!',
+                text: 'Your profile has been updated.',
+                icon: 'success',
+                confirmButtonText: 'Okay',
+              });
+            navigate('/profile')
+        }
+      
+    })
+
+    const data = new Date(userData.date_of_birth);
+    const date = `${data.getUTCFullYear()}-${String(data.getUTCMonth() + 1).padStart(2, '0')}-${String(data.getUTCDate()).padStart(2, '0')}`;
+
+    const handleChange = (name: string) => {
+        setFormErrors((prevState) => ({
+          ...prevState,
+          [name]: '',
+        }));
+      };
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        const formData: any = new FormData(e.currentTarget);
```

Comment:
Przyszły projekt robisz z react hook form, bo jak nie to masz wpierdol 😉. Tutaj wybaczalne, bo pierwszy projekt w react'cie i tak dużo rzeczy  nowych dochodzi to może być bez, ale ta obsługa formów meeega słabo developer friendly

Reply:
Dobra, next projekt React Hook Form!